src/utils/hooks/useListApplications.tsx

```diff
@@ -17,32 +16,36 @@ const useListApplication = () => {
   })
 
   useEffect(() => {
-    if (apolloError) return
-    if (apolloLoading) return
-    // Check that only one template matched
-    let error = checkForApplicationsErrors(data)
-    if (error) {
-      setError(error)
-      setLoading(false)
+    if (apolloError) {
+      setError(apolloError.message)
       return
     }
+    if (data && data.applications) {
+      const applicationsList = data?.applications?.nodes as Application[]
+      const applicationsDetails: ApplicationDetails[] = applicationsList.map((application) => {
+        const { id, serial, name, stage, status, outcome, template } = application
+        const stages = application.applicationStageHistories.nodes as ApplicationStageHistory[]
+        let stageId = -1
```

Comment:
Just because the field was expecting a number. If I use `undefined` I'll have to check elsewhere if the `stageId` isn't undefined. But not problem I can change that.

Reply:
Changed now. I ended moving the whole call for `useLoadApplication` to be inside this hook, and remove the check for always pass the `applicationId` and `stageId` so now I don't need to use a special value for before the application is loaded.